scalafix-rules/src/main/scala/scalafix/internal/rule/ExplicitResultTypes.scala

```diff
@@ -57,11 +69,21 @@ final class ExplicitResultTypes(
           )
         }
       }
-    if (config.scalacClasspath.nonEmpty && config.scalaVersion != supportedScalaVersion) {
+    if (config.scalacClasspath.nonEmpty && !supportedScalaVersions.contains(
+        config.scalaVersion
+      )) {
+      val shortScalaVersion =
+        config.scalaVersion.split(".").take(2).mkString(".")
+      val shortRuleScalaVersion =
+        Properties.versionNumberString.split(".").take(2).mkString(".")
       Configured.error(
-        s"The ExplicitResultTypes rule only supports the Scala version '$supportedScalaVersion'. " +
-          s"To fix this problem, either remove `ExplicitResultTypes` from .scalafix.conf or change the Scala version " +
-          s"in your build to match exactly '$supportedScalaVersion'."
+        s"Your scalaVersion is ${config.scalaVersion} but the ExplicitResultTypes rule is compiled with ${shortRuleScalaVersion}" +
+          s"and only supports these Scala versions '$supportedScalaVersions'. " +
+          (if (shortScalaVersion == shortRuleScalaVersion)
+             s"To fix this problem, either remove `ExplicitResultTypes` from .scalafix.conf or change the Scala version to be within '$supportedScalaVersions'"
+           else
+             s"To fix this problem, either remove `ExplicitResultTypes` from .scalafix.conf or " +
+               s"specify a `scalafixScalaBinaryVersion` key in your build.sbt equal to  $shortScalaVersion")
```

Comment:
For clarity, I think this message could be different depending on whether the binary version matches or not.

#### inputBinaryScalaVersion == runtimeBinaryScalaVersion
> The ExplicitResultTypes rule only supports the exact Scala versions '$supportedScalaVersion' for this binary version.  To fix this problem, either remove `ExplicitResultTypes` from `.scalafix.conf` or change the Scala version of your build to match one of the supported versions.

($supportedScalaVersion should list the full versions supported for the binary version $inputBinaryScalaVersion)

#### inputBinaryScalaVersion != runtimeBinaryScalaVersion
>The ExplicitResultTypes rule needs to run with the same Scala binary version as the one used to compile target sources ($inputBinaryScalaVersion). To fix this problem, either remove `ExplicitResultTypes` from `.scalafix.conf` or make sure the `scalafixScalaBinaryVersion` setting key matches  $inputBinaryScalaVersion.

Reply:
ok